hack/k8s-codegen.sh

```diff
@@ -0,0 +1,241 @@
+#!/usr/bin/env bash
```

Comment:
note: this file is almost the same as `hack/update-codegen.sh` but has enough little tweaks that it made sense to create a new file. It also also has the verification stuff built in now, so the name needed to change anyway.

Reply:
I'd really prefer to keep the consistent naming style of `update-*` and `verify-*` for different areas, but that's just my 2c... why can't we keep these scripts functioning even if it uses `make`? We're changing the developer experience rather than just bazel->make

Reply:
Perhaps move this to be something like `hack/lib/k8s-codegen.sh` and then having `update-codegen.sh` and `verify-codegen.sh` call into `k8s-codegen.sh` with the appropriate arguments?

Reply:
I didn't really see any value in keeping those scripts when the goal is that people use make for this kind of thing.

If we keep `update-codegen.sh` around and it references bazel then it's just another task to have to update that later when removing bazel, so I was hoping to save having to do that later.

Reply:
(following on from previous message)

and if we keep `update-codegen.sh` and it _doesn't_ reference bazel, then it's just a huge pain-in-the-ass alternative compared with `make update-codegen`

Plus without further changes to bazel (which I really don't want to do) `pull-cert-manager-bazel` will do the verify-codegen step as well as `pull-cert-manager-make-test`, and we end up duplicating the job until we remove bazel

Reply:
I think I may be misunderstanding why it's a lot of effort to have this file still around 🤔 but to answer a few points,

1) I wasn't expecting the new update-codegen.sh to still reference Bazel - rather that it will be the thing that invokes the `k8s-codegen.sh` script, and the Makefile would reference that script (i.e. `make update-codegen` calls `./hack/update-codegen.sh` which in turn calls this new file with the appropriate flags/arguments)

2) Perhaps I'm misunderstanding some of the complexity, but I'm not sure why it's a huge PITA alternative or what you mean by that (I think maybe we're misaligned around how much work is involved in this..)

3) The changes to Bazel in question need to be made anyway 😄 (notably, removing these two targets: https://github.com/cert-manager/cert-manager/blob/7dd845a2982958c5373503ec831ab934de515a8d/hack/BUILD.bazel#L137-L206)

It's not a hill I'm willing to die on, but I'm just personally very used to invoking various `update` and `verify` steps by invoking `./hack/update-thing.sh` as part of my development workflow :-) I wouldn't even be against keeping these files even if they just invoked the appropriate `make` commands for me 😄